### PR TITLE
Added stat_identity options for the pie chart

### LINKED ISSUE #5464: Adding a column chart button to the bar chart dialogue
I suggest this could be quite an easy addition.

The current Describe > Specific > Bar Chart dialogue has 2 buttons at the top.  I suggest a third button Column Chart after the Pie Chart.

This is the same as the Bar Chart, but with the stat="identity" as automatic.  It is alternatively geom_col which we have already.  So it can call geom_col.  
a) This is just the same as geom_bar, but with the stat="identity".  So in the properties of the geom I suggest we swap the order.  Currently the order is count, followed by identity - which is correct for geom bar.  I suggest it should be the other way round for geom_col.
b) On the main dialogue There should now be a new receiver.  I assume this should be the usual Y variable - which can therefore be Single/Multiple as for other graphics dialogues.  The existing single receiver is now the X-Variable.
c) Currently this x-variable can take numeric (as well as factors), though it is usually a factor.  Please keep this flexibility, i.e. don't restrict it to a factor.
d) While there, the pie chart option is restricted to a factor column.  Allow this to be (as for the ordinary bar chart) into numeric as well.
e) I suggest the third option could usefully also have the width of the bars on the main dialogue.  We will often want that to be much less than the default of 0.9.    If necessary that can come later.  

The issues of labeling are the same as in #5398 , except that labeling of an ordinary bar chart is just a count, while labeling here needs formatting decisions.  That can come later.

Here is an example of this third option:
image

### PR DESCRIPTION
**Pull request by Ogik99**
@africanmathsinitiative/developers this is ready for review.

fixes #5464
This implements the "stat_identity" option for the pie chart.

### CODE REVIEW

File: instat/dlgBarAndPieChart.vb [hunk 1]
```diff
@@ -78,11 +78,11 @@ Public Class dlgBarAndPieChart
...
         ucrReceiverByFactor.SetLinkedDisplayControl(lblByFactor)
-        ucrPnlOptions.AddToLinkedControls(ucrReceiverY, {rdoBarChart}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
-        ucrReceiverY.SetLinkedDisplayControl(lblYvariable)
+        'ucrPnlOptions.AddToLinkedControls(ucrReceiverY, {rdoBarChart}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
+        'ucrReceiverY.SetLinkedDisplayControl(lblYvariable)
 
-        ucrPnlOptions.AddToLinkedControls({ucrInputYValue}, {rdoBarChart}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
-        ucrInputYValue.SetLinkedDisplayControl(lblYValue)
+        'ucrPnlOptions.AddToLinkedControls({ucrInputYValue}, {rdoBarChart}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
+        'ucrInputYValue.SetLinkedDisplayControl(lblYValue)
```
Comment (dannyparsons) [inline instat/dlgBarAndPieChart.vb]: Should these be removed if they're not needed any more?

**Comment by rdstern** [comment]
I assume by "stat identity" you mean the option for the pie chart to use a variable - as is done for the bar chart.
If so, then this works.  
The only issue  - that is different from the bar chart is that when you choose the option "Variable", then it correctly opens the control for the variable.  But OK is enabled - which it isn't for the bar chart in the same way.  So you can press OK without giving a variable and that generates an error.  So just to sort out that small point and it can be merged.

**Comment by shadrackkibet** [comment]
@Ogik99 did you sort out the second point from @rdstern above? any help needed?

**Comment by lloyddewit** [comment]
@shadrackkibet I have a favour to ask. @Ogik99 is not currently active on R-Instat and it would be nice to close this PR. Please could you resolve the last issue identified by @rdstern so that we can merge? thanks if this is possible

**Comment by lloyddewit** [comment]
@shadrackkibet Please ignore my request above. I noticed that @Ogik99 has a number of open PRs. I will look at all these together to see how best to resolve. 
But be warned, I may ask for your help again later ;)

**Comment by lloyddewit** [comment]
@Ivanluv Please could you try and complete this PR? (@rdstern suggested that you would be the best person for this task)

**Comment by lloyddewit** [comment]
@rdstern I had to make some small changes to resolve merge conflicts. Therefore your approval is needed again. The merge conflicts only affected the position of the controls on the dialog. If the dialog's appearance looks OK to you then it is probably safe to approve, thanks

**Comment by rdstern** [comment]
@Ivanluv the changes made have resulted in a slight problem with the position label and corresponding control.  Once that is resolved (again!) this can be merged.
image

**Comment by Ivanluv** [comment]
Am looking at this

**Pull request merged**